### PR TITLE
feat(eslint-plugin-lit-a11y): add heading-has-content function calls

### LINKED ISSUE #1879: [eslint-plugin-lit-a11y] heading-has-content support function calls
The following code is valid code, but currently fails the linting rule:

```js
html`<h1>${foo()}</h1>`
```

This, however, does pass:
```js
html`<h1>${foo}</h1>`
```

Gotta figure out why the former fails, and fix it

You can add a failing test here:

https://github.com/open-wc/open-wc/blob/34ef9e392b8c3a21dc75c809d8bcdafab7aa885b/packages/eslint-plugin-lit-a11y/tests/lib/rules/heading-has-content.js#L24

By adding the following line:

```js
    { code: "html`<h2>${foo()}</h2>`" },
```

Then for a smooth debug experience you can run: 
```bash
mocha tests/lib/rules/heading-has-content.js --watch
```

### LINKED ISSUE #1880: [eslint-plugin-lit-a11y] aria-attr-valid-value support function calls
Seems to be a similar problem as https://github.com/open-wc/open-wc/issues/1879, see that issue for more info

Failing test:
```js
    { code: 'html`<div aria-label="${foo("foo")}"></div>`' },
```

### PR DESCRIPTION
**Pull request by trystan2k**
Hi again :D

This one should fix #1879 

Please let me know if you see any issue !

Thanks !

### CODE REVIEW

**Comment by trystan2k** [comment]
I just realize that this also closes #1880 as it share the same `utils.js` file. 
I added a new commit with the test from the issue.

File: packages/eslint-plugin-lit-a11y/tests/lib/rules/aria-attr-valid-value.js [hunk 1]
```diff
@@ -109,6 +109,9 @@ ruleTester.run('aria-attr-valid-value', rule, {
     { code: 'html`<div aria-labelledby="additions additions" />`' },
     { code: 'html`<div aria-labelledby="additions removals text" />`' },
     { code: 'html`<div aria-labelledby="additions removals text all" />`' },
+
+    // FUNCTION CALL
+    { code: 'html`<div aria-label="${foo("foo")}"></div>`' },
```
Comment (thepassle) [inline packages/eslint-plugin-lit-a11y/tests/lib/rules/aria-attr-valid-value.js:114]: Could you add a test for:
```js
    { code: 'html`<div aria-label="${this.foo("foo")}"></div>`' },
```

as well? Just to make sure 🙂

**Comment by thepassle** [comment]
I think this should also fix https://github.com/open-wc/open-wc/issues/1880

Could you add the (currently) failing test from that issue and see if its solved as well?

File: packages/eslint-plugin-lit-a11y/tests/lib/rules/aria-attr-valid-value.js (hunk 1 above)
Comment (trystan2k) [reply packages/eslint-plugin-lit-a11y/tests/lib/rules/aria-attr-valid-value.js:114]: Sure. 
Just added and test passed as well :D

**Comment by trystan2k** [comment]
> I think this should also fix #1880
> 
> Could you add the (currently) failing test from that issue and see if its solved as well?

I think it was the 2 tests I added for `aria-attr-valid-value.js`, right ? I tried to link the PR to 2 issues (not sure if it is doable) and could not. But yes, in theory this PR fix both issues.

File: packages/eslint-plugin-lit-a11y/tests/lib/rules/heading-has-content.js [hunk 2]
```diff
@@ -36,6 +36,12 @@ ruleTester.run('heading-has-content', rule, {
     { code: 'html`<h5>${foo}</h5>`' },
     { code: 'html`<h6>${foo}</h6>`' },
     { code: "html`<h1><div aria-hidden='true'>foo</div> foo</h1>`" },
+    { code: 'html`<h1>${foo()}</h1>`' },
+    { code: 'html`<h1>${foo("hello")}</h1>`' },
+    { code: 'html`<h1>${foo(1)}</h1>`' },
+    { code: 'html`<h1>${foo(true)}</h1>`' },
+    { code: 'html`<h1>${foo(bar)}</h1>`' },
+    { code: 'html`<h1>${foo(bar, "hello", 1, true)}</h1>`' },
```
Comment (thepassle) [inline packages/eslint-plugin-lit-a11y/tests/lib/rules/heading-has-content.js:44]: could you add one for `this.foo()` here as well?

**Comment by thepassle** [comment]
> I think it was the 2 tests I added for `aria-attr-valid-value.js`, right ? I tried to link the PR to 2 issues (not sure if it is doable) and could not. But yes, in theory this PR fix both issues.

Ah yes, my bad, looked too fast 🙂 Added one more comment 👍

File: packages/eslint-plugin-lit-a11y/tests/lib/rules/heading-has-content.js (hunk 2 above)
Comment (trystan2k) [reply packages/eslint-plugin-lit-a11y/tests/lib/rules/heading-has-content.js:44]: Done !

**Comment by trystan2k** [comment]
> > I think it was the 2 tests I added for `aria-attr-valid-value.js`, right ? I tried to link the PR to 2 issues (not sure if it is doable) and could not. But yes, in theory this PR fix both issues.
> 
> Ah yes, my bad, looked too fast 🙂 Added one more comment 👍

No problem !

File: packages/eslint-plugin-lit-a11y/tests/lib/rules/heading-has-content.js (hunk 2 above)
Comment (trystan2k) [reply packages/eslint-plugin-lit-a11y/tests/lib/rules/heading-has-content.js:44]: Sorry, just saw I did not done correctly. Just pushed it correctly now

**Review by thepassle — APPROVED** [review]

**Comment by thepassle** [comment]
fixes https://github.com/open-wc/open-wc/issues/1880
fixes https://github.com/open-wc/open-wc/issues/1879

**Comment by thepassle** [comment]
Thanks again for the great work 🙂

**Pull request merged**